discovery-provider/tests/conftest.py

```diff
@@ -74,54 +74,103 @@ def celery_config():
     }
 
 
-@pytest.fixture
+@pytest.fixture(scope='module')
 def celery_app():
+    """
+    Configures a test fixture for celery.
+    Usage:
+    ```
+    def test_something(celery_app):
+        task = celery_app.celery.tasks["update_something"]
+        db = task.db
+        with db.scoped_session():
+            pass
+    ```
+
+    Note: This fixture must be at module scope because celery
+    works by autodiscovering tasks with a decorator at import time of a module.
+    If you try to use celery as a fixture at the function scope,
+    you will run into mysterious errors as some task context may be stale!
+    """
+    # Drop DB, ensuring migration performed at start
```

Comment:
maybe we should do [teardown after the `yield`](https://docs.pytest.org/en/6.2.x/fixture.html#yield-fixtures-recommended)?

Reply:
I was trying to do this initially for both celery_app and app fixture, but you run into an issue where you need each test to adhere to using the fixture or another test that *doesn't* do the cleanup at the beginning will break. I do want to see into making this change, but I figured it would be way way way too big for this PR

Reply:
makes sense. thanks for explaining your reasoning. i agree that it's out-of-scope for this PR.